src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs

```diff
@@ -3762,6 +3771,14 @@ void addProperty(SynthesizedRecordPropertySymbol property)
                 }
 
                 return existingOrAddedMembers.ToImmutableAndFree();
+
+                void validateNotHidden(Symbol symbol, ParameterSymbol param)
+                {
+                    if (memberNames.Contains(symbol.Name) || !this.GetTypeMembers(symbol.Name).IsEmpty)
```

Comment:
:memo: I have to say I'm a bit worried about this. The next feature we add where it turns out that computing type-members depends on computing non-type-members will cause a cycle.

Reply:
>  I have to say I'm a bit worried about this. The next feature we add where it turns out that computing type-members depends on computing non-type-members will cause a cycle.

I don't think this adds any new possible exposure. We are binding signatures of members in ```AddSynthesizedRecordMembersIfNecessary``` and those can refere to types declared inside the record.